tools/rewriter/DetermineAbi.cpp

```diff
@@ -204,32 +219,53 @@ cgFunctionInfo(clang::CodeGen::CodeGenModule &cgm,
   }
 }
 
+// Given a function signature, find the layout of the argument and return value slots.
 CAbiSignature determineAbi(const clang::CodeGen::CGFunctionInfo &info,
                            const clang::ASTContext &astContext) {
   // get ABI for return type and each parameter
   CAbiSignature sig;
   sig.variadic = info.isVariadic();
+
+  // We want to find the layout of the parameter and return value "slots."
+  // We can store a certain number of slots in registers, while the rest
+  // will go on the stack.
+  // These are "slots" and not parameters because some parameters larger
+  // than 64 bits can occupy multiple slots.
+  // Each slot is of one of the types described in CAbiArgKind.
+
+  // Get the slots for the return value.
   auto &returnInfo = info.getReturnInfo();
   sig.ret = abiSlotsForArg(info.getReturnType(), returnInfo, astContext);
 
   auto is_integral = [](auto &x) { return x == CAbiArgKind::Integral; };
+
+  // num_regs is the number of registers in which the return value is stored.
+  // This may be one for a value up to 64 bits or two for a 128-bit value.
```

Comment:
This is specifically only counting the number of integral registers--floating-point registers (`xmmN`) will not be counted here, and I'm not sure if that's a bug. I believe @ayrtonm originally wrote this code (see #86), and it never appears to have counted float-kind eightbytes.